pkg/checks/traceroute/traceroute.go

```diff
@@ -38,9 +41,14 @@ func randomPort() int {
 	return rand.N(portRange) + basePort // #nosec G404 // math.rand is fine here, we're not doing encryption
 }
 
-func tcpHop(ctx context.Context, addr net.Addr, ttl int, timeout time.Duration) (net.Conn, int, error) {
+// tcpHop attempts to connect to the target host using TCP with the specified TTL and timeout.
+// It returns a [net.Conn], the port used for the connection, and an error if the connection failed.
+func tcpHop(ctx context.Context, addr net.Addr, ttl int, timeout time.Duration) (conn net.Conn, port int, err error) {
```

Comment:
Is there really a need for the named returns? If there really is, then we should consider refactoring this function. Seems OK without the named returns.

Reply:
I've just added them for documentation but if you think it makes it less readable, I can take them out again.

Reply:
We've already discussed this in previous occasions; named returns are a gotcha in go we don't want to mess with. Why roll the dice when there's no need to? The function is already documented. I'll let @niklastreml decide on that, as he's written this code.

Please keep in mind, that we're not using named returns elsewhere in the codebase. Consistency is also an important aspect of a solid codebase ;)

Reply:
Let's remove the named returns, the comments on the function should be documentation enough.